Warzone/src/GameEngine.cpp

```diff
@@ -177,17 +177,31 @@ Map* GameEngine::chooseMap() {
     bool validMap = 0;
     Map* map = NULL;
     while(validMap != 1) {
-        cout << "Which file do you want to use as the map?\n\n";
+        cout << "Which file do you want to use as the map?\n";
         string filePath;
         cin >> filePath;
-        MapLoader* mapL = new MapLoader(filePath); // load the file
-        validMap = mapL->parse(); // check if it is a valid map
-        map = mapL->createMap(); // get the map
-
-        // delete pointer and dangling pointer
-        delete mapL;
-        mapL = NULL;
+        // if file name contains Conquest_
+        if (filePath.find("Conquest_") != string::npos) {
+            // use the adapter to read conquest maps
+            ConquestFileReader *conquestLoader = new ConquestFileReader(filePath);
+            ConquestFileReaderAdapter *adapter = new ConquestFileReaderAdapter(conquestLoader);
+            validMap = adapter->parse();
+            cout << "Creating map..." << endl;
+            map = adapter->createMap();
```

Comment:
Trying to load a Conquest map through GameEngine stops here: 
![image](https://user-images.githubusercontent.com/35614258/100664553-ccefb480-3325-11eb-8d0b-29880d04839c.png)